tee-worker/litentry/core/parachain-extrinsic-task/receiver/src/lib.rs

```diff
@@ -0,0 +1,56 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+use itp_extrinsics_factory::CreateExtrinsics;
+use itp_ocall_api::EnclaveOnChainOCallApi;
+use itp_types::parentchain::ParentchainId;
+use lc_parachain_extrinsic_task_sender::init_parachain_extrinsic_sender_storage;
+use log::*;
+use std::{format, string::String, sync::Arc, time, vec};
+
+const MAX_BATCH_SIZE: usize = 500;
+const BATCH_EXTRINSIC_INTERVAL: time::Duration = time::Duration::from_secs(6);
+
+pub fn run_parachain_extrinsic_task_receiver<ExtrinsicsFactory, OCallApi>(
+	api: Arc<OCallApi>,
+	extrinsic_factory: Arc<ExtrinsicsFactory>,
+) -> Result<(), String>
+where
+	ExtrinsicsFactory: CreateExtrinsics + Send + Sync + 'static,
+	OCallApi: EnclaveOnChainOCallApi,
+{
+	let task_receiver = init_parachain_extrinsic_sender_storage().map_err(|e| {
+		format!("Failed to initialize parachain extrinsic task sender storage: {:?}", e)
+	})?;
+	let mut calls = vec::Vec::new();
+
+	loop {
+		let start_time = time::Instant::now();
+		while start_time.elapsed() < BATCH_EXTRINSIC_INTERVAL && calls.len() < MAX_BATCH_SIZE {
```

Comment:
So I went through the logic, this particular condition - `start_time.elapsed() < BATCH_EXTRINSIC_INTERVAL && calls.len() < MAX_BATCH_SIZE` , it would suggest that at high traffic we will still be submitting a lot of extrinsics but in the form of batch extrinsics, so batches of 500 at a time correct me if I'm wrong (which more or less occupies the same weight as all the calls). This will still take up a lot of the block weight and not allow more important calls like `ParentchainBlockProcessed` to be present in the block, and yes the nonce increments will be fewer since we are submitting in batches. 

So we need to decouple these two conditions, We create a vector of all batched calls like this - `all_batched_calls: Vec<BoundedVec<500>>`. So when we hit capacity we insert the calls into it probably like this `all_batched_calls.insert(calls)`, and only when we hit the interval time, we submit these calls. So this max size should be such that it should take up only half of the block weight in the worst case, so that the chain is usable for other purposes and not spammed with worker extrinsics during high traffic. I'm just suggesting one logic, you can do this however you like.